src/window/basic_frame.rs

```diff
@@ -543,26 +558,24 @@ impl Frame for BasicFrame {
         if self.hidden {
             (width, height)
         } else {
-            (
-                width - 2 * (DECORATION_SIZE as i32),
-                height - DECORATION_SIZE as i32 - DECORATION_TOP_SIZE as i32,
-            )
+            (width, height - DECORATION_TOP_SIZE as i32)
         }
     }
 
     fn add_borders(&self, width: i32, height: i32) -> (i32, i32) {
         if self.hidden {
             (width, height)
         } else {
-            (
-                width + 2 * (DECORATION_SIZE as i32),
-                height + DECORATION_SIZE as i32 + DECORATION_TOP_SIZE as i32,
-            )
+            (width, height + DECORATION_TOP_SIZE as i32)
         }
     }
 
-    fn location(&self) -> (i32, i32) {
-        (-(DECORATION_SIZE as i32), -(DECORATION_TOP_SIZE as i32))
+    fn headless_geometry(&self, x: i32, y: i32, width: i32, height: i32) -> (i32, i32, i32, i32) {
```

Comment:
This trait is a public interface, so adding a new required method to it is a breaking change.

I'm not sure I see why this method is needed though, couldn't you have reached the same result by changing the implementation of `location()` to this?

```rust
fn location(&self) -> (i32, i32) {
    if self.hidden { (0, 0) } else { (0, -DECORATION_TOP_SIZE as i32) }
}
```